pkg: add Honor and Huawei packages

AnonymousWP opened the pull request:
Fixes #520. Fixes #518. Fixes #624.

AnonymousWP committed:
Add Honor and Huawei packages

AnonymousWP committed:
Add leftover Huawei and Honor packages

AnonymousWP committed:
Add packages and change removals

AnonymousWP commented:
Maybe better to continue our discussion here, @VistHJ and @kyoforkshomebrews.

VistHJ commented:
I've got some information:

- `com.huawei.android.internal.app`
> Component of Huawei sharing. **I read someone saying "Do not remove or you won't be able to send files to apps"**.\nCan someone test it ? \n
**Myth busted,** here's proof: [https://youtu.be/RDEnyPl78XI]

- ### com.huawei.android.internal.app
> This is a part of the PowerGenie, the task killer app. If you want to remove it, uninstall `com.huawei.powergenie` along the way

kyoforkshomebrews commented:
Is it still possible to use the share menu though? Because i think that's what ```com.huawei.android.internal.app``` does? The video only shows the file selector.

AnonymousWP committed:
Edit descriptions regarding feedback

AnonymousWP commented:
> I've got some information:
> 
> * ### com.huawei.android.internal.app
> 
> > Component of Huawei sharing. **I read someone saying "Do not remove or you won't be able to send files to apps"**.\nCan someone test it ? \n
> > `Myth busted, here's proof: https://youtu.be/RDEnyPl78XI`
> 
> * ### com.huawei.android.hwaps
> 
> > This is a part of the PowerGenie, the task killer app. If you want to remove it, uninstall `com.huawei.powergenie` along the way

Done, please check.

jorants commented:
On my phone many of the packages under `com.huawei.***` have been renamed `com.hihonor.***` . Apart from that they are the same package. Including both would simply require duplicating the existing entries, the following python script worked for me:

```python
import json, sys

def make_honor(app):
    # dont want to change the previos one:
    app = app.copy()
    app['id'] = app['id'].replace("com.huawei.","com.hihonor.")
    if app['dependencies'] is not None:
        app['dependencies'] = [name.replace("com.huawei.","com.hihonor.") for name in app['dependencies']]
    if app['neededBy'] is not None:
        app['neededBy'] = [name.replace("com.huawei.","com.hihonor.") for name in app['neededBy']]
    return app

with open(sys.argv[1]) as fp:
    apps = json.load(fp)
    
apps = [*apps,
        *(make_honor(app) for app in apps if app['id'].startswith('com.huawei.'))]

with open(sys.argv[2],"w") as fp:
    json.dump(apps,fp,indent=4)
```
I'm happy to make a pull request as soon as this one has been merged.

kyoforkshomebrews commented:
I want to also add following info:

### com.huawei.hwdockbar
**List**: `Oem`
**Removal**: `Advanced`
> Huawei Multi-Window dock. Probably fine to remove if you're not using Huawei Multi-Window features.

### com.huawei.filemanager
**List**: `Oem`
**Removal**: `Advanced`
> Huawei file manager. Probably fine to remove as long as you have another file manager.

AnonymousWP committed:
Add three Huawei packages

`com.huawei.hwdockbar`, `com.huawei.filemanager` and `com.huawei.hms5gkit.agentservice`

AnonymousWP commented:
> 

Thanks, done (see commit).

AnonymousWP committed:
pkg: add Honor Honor 10 Lite packages

Fixes #624

kyoforkshomebrews commented:
com.huawei.android.dsdscardmanager` also exists as com.huawei.dsdscardmanager`

kyoforkshomebrews commented:
### com.huawei.permissioncontroller
**List**: `Oem`
**Removal**: `Recommended`
> Can be disabled via Settings.

https://forum.xda-developers.com/t/guide-remove-huawei-bloatware-without-root-access-using-adb-usb-debugging.4014937/page-19#post-86890689
According to this source the following have been added in EMUI 12 and can be removed. I also tried disabling them on my phone and it seems that it's fine:

`
com.huawei.android.airsharing
com.huawei.contentsensor
com.huawei.devicemanager
com.huawei.hms5gkit.agentservice
com.huawei.hwdiagnosis
com.huawei.ohos.hiwindow
com.huawei.onehopsvcclient
com.huawei.onehopsvchost
com.huawei.profile
com.huawei.trustedthingsauth
`

### com.huawei.meetime
**List**: `Oem`
**Removal**: `Recommended`
> Meetime video calling app. Can be removed via settings.

### com.huawei.mediacontroller
**List**: `Oem`
**Removal**: `Advanced`
> Can be disabled via Settings. Will disable the media controls in quick settings. Won't remove them however.

AnonymousWP committed:
pkg: add more Huawei packages

AnonymousWP commented:
> The package `com.huawei.android.dsdscardmanager` also exists as `com.huawei.dsdscardmanager`

So, same package, different package name and same purpose (including list, description and removal)?

> https://forum.xda-developers.com/t/guide-remove-huawei-bloatware-without-root-access-using-adb-usb-debugging.4014937/page-19#post-86890689 According to this source the following have been added in EMUI 12 and can be removed. I also tried disabling them on my phone and it seems that it's fine:
> 
> ```
> com.huawei.android.airsharing
> com.huawei.contentsensor
> com.huawei.devicemanager
> com.huawei.hms5gkit.agentservice
> com.huawei.hwdiagnosis
> com.huawei.ohos.hiwindow
> com.huawei.onehopsvcclient
> com.huawei.onehopsvchost
> com.huawei.profile
> com.huawei.trustedthingsauth
> ```

Would you be able to find out the description, removal and list that's required for this project? I have added the ones you listed.

> com.huawei.meetime
> List: Oem
> Removal: Recommended
> 
> Meetime video calling app. Can be removed via settings.

This one was already added.

kyoforkshomebrews commented:
I suppose so. Is there any way to try the updated uad-list out? I want to check what's still missing.

AnonymousWP commented:
> I suppose so. Is there any way to try the updated uad-list out? I want to check what's still missing.

No, unless you compile it with the new list. You can view my updated list here: https://github.com/AnonymousWP/universal-android-debloater/blob/520-Honor-unlisted-packages/resources/assets/uad_lists.json

kyoforkshomebrews commented:
How do I do that? I compiled it after replacing /resources/assets/uad_lists.json but it seems that didn't work.

AnonymousWP commented:
> How do I do that? I compiled it after replacing /resources/assets/uad_lists.json but it seems that didn't work.

Hmm, not sure. Have you followed https://github.com/0x192/universal-android-debloater/wiki#build ?

kyoforkshomebrews commented:
> > How do I do that? I compiled it after replacing /resources/assets/uad_lists.json but it seems that didn't work.
> 
> Hmm, not sure. Have you followed https://github.com/0x192/universal-android-debloater/wiki#build ?

Yes, I tried running my compiled version without internet access but it still doesn't change the data shown.

AnonymousWP commented:
> > > How do I do that? I compiled it after replacing /resources/assets/uad_lists.json but it seems that didn't work.
> > 
> > 
> > Hmm, not sure. Have you followed https://github.com/0x192/universal-android-debloater/wiki#build ?
> 
> Yes, I tried running my compiled version without internet access but it still doesn't change the data shown.

I don't really know then. Haven't tried it myself. Did you find a solution by now?

kyoforkshomebrews commented:
Yes. I had to also delete ~/.cache/uad/uad_lists.json

Anyways I think this can be merged.
@AnonymousWP

AnonymousWP commented:
This repo is dead, so that's why we revived it, see https://github.com/Universal-Debloater-Alliance/universal-android-debloater. I think the commits in this PR should be in that repo, I just compared.
